### PR TITLE
Fix #17108 - Add JSON output for variable xrefs

### LINKED ISSUE #17108: Add JSON support for variable xref (afvx, afvR, afvW)
**Is your feature request related to a problem? Please describe.**

The commands `afvx`, `afvR`, `afvW` and `axv`, shows references to a variable inside the function:
Currently, the output of the variable references isn't supporting the JSON format.

```
[0x0804846b]> afvx
afvR
    arg_8h  0x8048471,0x804848a,0x8048499,0x80484a8,0x80484b0,0x80484c5,0x80484d4,0x80484e3,0x80484eb,0x80484fd
    var_4h  0x8048487,0x8048496,0x80484a5,0x80484ad,0x80484c2,0x80484d1,0x80484e0,0x80484e8,0x80484fa,0x8048509,0x8048518
afvW
    arg_8h
    var_4h  0x804847b
```

**Describe the solution you'd like**

It would be great to have these commands support the JSON output by appending the `j` letter to each command

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Closing issues**

Closes #17108

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -976,21 +976,38 @@ static void var_help(RCore *core, char ch) {
...
-	r_cons_printf ("%10s", name);
+	if (!pj) {
+		r_cons_printf ("%10s", name);
+	}
 	r_vector_foreach (&var->accesses, acc) {
 		if (!(acc->type & access_type)) {
 			continue;
 		}
-		r_cons_printf ("%s0x%"PFMT64x, first? "  ": ",", (ut64)((st64)fcn->addr + acc->offset));
+		ut64 addr = (ut64)((st64)fcn->addr + acc->offset);
```
Comment (trufae) [inline libr/core/cmd_anal.c:989]: why this cast

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -976,21 +976,38 @@ static void var_help(RCore *core, char ch) {
...
+		if (pj && first) {
+			pj_o (pj);
+			pj_ks (pj, "name", name);
+			pj_ka (pj, "addrs");
+		}
+		if (pj) {
+			pj_s (pj, sdb_fmt ("%"PFMT64u, addr));
```
Comment (trufae) [inline libr/core/cmd_anal.c:996]: use pj_n(pj, addr);

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -1242,7 +1280,15 @@ static int var_cmd(RCore *core, const char *str) {
...
 			}
-			list_vars (core, fcn, str[0], name);
+			PJ *pj = NULL;
+			if (str[1] == 'j') {
+				pj = pj_new ();
+			} 
+			list_vars (core, fcn, pj, str[0], name);
+			if (str[1] == 'j') {
+				pj_end (pj);
+				r_cons_printf ("%s\n", pj_drain (pj));
```
Comment (trufae) [inline libr/core/cmd_anal.c:1290]: memory leak here the result of pj_drain must be free'd

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -976,21 +976,38 @@ static void var_help(RCore *core, char ch) {
...
+	if (!pj) {
+		r_cons_printf ("%10s", name);
+	}
 	r_vector_foreach (&var->accesses, acc) {
 		if (!(acc->type & access_type)) {
 			continue;
 		}
-		r_cons_printf ("%s0x%"PFMT64x, first? "  ": ",", (ut64)((st64)fcn->addr + acc->offset));
+		ut64 addr = fcn->addr + acc->offset;
+		if (pj && first) {
```
Comment (ret2libc) [inline libr/core/cmd_anal.c:990]: why does it create a new json object on the first element and then never closes it? i would have expected that all variable accesses were in a array.

Comment (GustavoLCR) [reply libr/core/cmd_anal.c]: @ret2libc I don't know why I did that, fixed it 4e1e11a24b5f320e49aa350d3254ec3bf199fefe

**Pull request merged**